cmd/thanos/store.go

```diff
@@ -40,99 +43,144 @@ import (
 	"github.com/thanos-io/thanos/pkg/ui"
 )
 
-// registerStore registers a store command.
-func registerStore(app *extkingpin.App) {
-	cmd := app.Command(component.Store.String(), "Store node giving access to blocks in a bucket provider. Now supported GCS, S3, Azure, Swift, Tencent COS and Aliyun OSS.")
+type storeConfig struct {
+	indexCacheConfigs           extflag.PathOrContent
+	objStoreConfig              extflag.PathOrContent
+	dataDir                     string
+	grpcConfig                  grpcConfig
+	httpConfig                  httpConfig
+	indexCacheSizeBytes         units.Base2Bytes
+	chunkPoolSize               units.Base2Bytes
+	maxSampleCount              uint64
+	maxTouchedSeriesCount       uint64
+	maxConcurrency              int
+	component                   component.StoreAPI
+	debugLogging                bool
+	syncInterval                time.Duration
+	blockSyncConcurrency        int
```

Comment:
Can we ensure we are reusing configs from other components? (:

Reply:
Apart from `webConfig` (Since we dont use prefix route in for store command), Yes. Also this was mentioned [here](https://github.com/thanos-io/thanos/pull/2740#discussion_r437138647)